rpcs3/Emu/RSX/VK/VKTextureCache.h

```diff
@@ -890,8 +889,8 @@ namespace vk
 							range_reset = true;
 						}
 
-						// Upgrade to writer lock
-						lock.upgrade();
+						//Warning: Will deadlock if a double fault occurs and we enter here from 'flush' method
```

Comment:
We never acquire a writer lock when handling any type of page fault so it should unwind ok for now. I'll have to find a better way of predetermining all adjacent blocks that will interfere with one another first, then handle protection outside the flush itself.